includes/Integrations/Jetpack.php

```diff
@@ -128,10 +128,16 @@ public function filter_ajax_query_attachments_args( array $args ) {
 			return $args;
 		}
 
-		$allowed_mime_types = $this->get_allowed_mime_types();
-		$allowed_mime_types = array_merge( ...array_values( $allowed_mime_types ) );
-		if ( in_array( self::VIDEOPRESS_MIME_TYPE, $args['post_mime_type'], true ) && ! array_diff( $allowed_mime_types, $args['post_mime_type'] ) ) {
-			add_filter( 'wp_prepare_attachment_for_js', [ $this, 'filter_admin_ajax_response' ], 10, 2 );
+		$allowed_mime_types              = $this->get_allowed_mime_types();
+		$allowed_mime_types              = array_merge( ...array_values( $allowed_mime_types ) );
+		$allowed_mime_types_transcodable = array_merge( $allowed_mime_types, $this->get_allow_transcodable_mime_types() );
+		if ( in_array( self::VIDEOPRESS_MIME_TYPE, $args['post_mime_type'], true ) ) {
+			if ( ! array_diff( $allowed_mime_types, $args['post_mime_type'] ) ) {
+				add_filter( 'wp_prepare_attachment_for_js', [ $this, 'filter_admin_ajax_response' ], 10, 2 );
+			}
+			if ( ! array_diff( $allowed_mime_types_transcodable, $args['post_mime_type'] ) ) {
+				add_filter( 'wp_prepare_attachment_for_js', [ $this, 'filter_admin_ajax_response' ], 10, 2 );
+			}
```

Comment:
It is impossible to know if transcoding is possible as this is at broswer level. So check, with and without transcodable mime types.